MaterialDesignThemes.Wpf/PaletteHelper.cs

```diff
@@ -123,23 +123,35 @@ public void ReplaceAccentColor(string name)
         /// <param name="newValue">The new entry value</param>
         /// <param name="parentDictionary">The root dictionary to start searching at. Null means using Application.Current.Resources</param>
         /// <returns>Weather the value was replaced (true) or not (false)</returns>
-        private static bool ReplaceEntry(object entryName, object newValue, ResourceDictionary parentDictionary = null)
+        private static bool ReplaceEntry(object entryName, object newValue, ResourceDictionary parentDictionary = null, bool animate = true)
         {
+            const int DURATION_MS = 500; //Change the value if needed
+            
             if (parentDictionary == null)
                 parentDictionary = Application.Current.Resources;
-
+            
             if (parentDictionary.Contains(entryName))
             {
-                parentDictionary[entryName] = newValue;
-                return true;
+                if (animate) //Fade animation is enabled
+                {
+                    ColorAnimation animation = new ColorAnimation()
+                    {
+                        From = (Color)parentDictionary[entryName],//The old color
+                        To = (Color)newValue, //The new color
+                        Duration = new Duration(new TimeSpan(0,0,0,0,DURATION_MS))
+                    };
+                    (parentDictionary[entryName] as Brush).BeginAnimation(SolidColorBrush.ColorProperty, animation); //Begin the animation
+                    return true;
```

Comment:
This ```return true``` shouldn't be just inside ```if (animate)```: Right now if we set ```animate``` to ```false``` your code will search the merged dictionaries and may even return ```false``` for the entry even though we already found it.

You should put it after line 149: Inside the ```if (parentDictionary.Contains(entryName))```, but outside the ```if (animate)``` and its ```else```.